FinSearchUnified/Tests/Bundle/SearchBundleFindologic/FacetHandler/CategoryFacetHandlerTest.php

```diff
@@ -0,0 +1,445 @@
+<?php
+
+namespace FinSearchUnified\Tests\Bundle\SearchBundleFindologic\FacetHandler;
+
+use FinSearchUnified\Bundle\SearchBundleFindologic\FacetHandler\CategoryFacetHandler;
+use Shopware\Bundle\SearchBundle\Condition\ProductAttributeCondition;
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Bundle\SearchBundle\Facet\ProductAttributeFacet;
+use Shopware\Bundle\SearchBundle\FacetResult\TreeFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\TreeItem;
+use Shopware\Bundle\SearchBundle\FacetResultInterface;
+use Shopware\Components\Test\Plugin\TestCase;
+use SimpleXMLElement;
+
+class CategoryFacetHandlerTest extends TestCase
+{
+    /**
+     * @dataProvider filterProvider
+     *
+     * @param string $name
+     * @param bool $doesSupport
+     */
+    public function testSupportsFilter($name, $doesSupport)
+    {
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+
+        $filter->addChild('name', $name);
+
+        $facetHandler = new CategoryFacetHandler();
+        $result = $facetHandler->supportsFilter($filter);
+
+        $this->assertSame($doesSupport, $result);
+    }
+
+    public function filterProvider()
+    {
+        return [
+            'Category filter' => ['cat', true],
+            'Vendor filter' => ['vendor', false],
+        ];
+    }
+
+    /**
+     * @dataProvider treeItemProvider
+     *
+     * @param array $filterData
+     * @param string $mode
+     * @param FacetResultInterface $facetResult
+     * @param ConditionInterface|null $condition
+     */
+    public function testGeneratesPartialFacetBasedOnFilterDataAndActiveConditions(
+        array $filterData,
+        $mode,
+        FacetResultInterface $facetResult,
+        ConditionInterface $condition = null
+    ) {
+        $facet = new ProductAttributeFacet('cat', $mode, 'cat', 'Category');
+
+        $criteria = new Criteria();
+        if ($condition !== null) {
+            $criteria->addCondition($condition);
+        }
+
+        $filter = $this->generateFilter($filterData);
+
+        $facetHandler = new CategoryFacetHandler();
+        $result = $facetHandler->generatePartialFacet($facet, $criteria, $filter);
+
+        $this->assertEquals($facetResult, $result);
+    }
+
+    public function treeItemProvider()
+    {
+        return [
+            'Facet with filter without condition' => [
+                // $filterData
```

Comment:
You can use array keys for this instead of comments e.g. `filterData => ['name` => 'cat',...]`